lib/dolly/document.rb

```diff
@@ -110,6 +110,19 @@ def self.property *ary
       end
     end
 
+    def method_missing method_name, *args
+      if method_name.to_s.chars.last == '='
+        raise ArgumentError unless args.first == args.last
+        puts args.first
+        doc["#{method_name.to_s.chop}"] = args.first
+      else
+        raise NoMethodError unless doc["#{method_name}"].present?
+        raise ArgumentError unless args.count.zero?
+
+        doc["#{method_name}"]
+      end
+    end
```

Comment:
Add `&block` to the signature.
To test for a trailing equal sign `method_name.include? '='` and `method_name =~ /=$/` are nicer imho.
Its usually a good idea to use `define_method` within `method_missing` so subsequent calls are defined.
If `doc["method_name"]` doesn't already exist call super so lookup can continue up the ancestor chain and return a proper stack trace.